tentacle/src/builder.rs

```diff
@@ -217,9 +218,33 @@ where
     #[cfg(not(target_family = "wasm"))]
     pub fn tcp_config<F>(mut self, f: F) -> Self
     where
-        F: Fn(TcpSocket) -> Result<TcpSocket, std::io::Error> + Send + Sync + 'static,
+        F: Fn(TcpSocket, TransformerContext) -> Result<TcpSocket, std::io::Error>
+            + Send
+            + Sync
+            + 'static,
     {
-        self.config.tcp_config.tcp = Arc::new(f);
+        self.config.tcp_config.tcp.socket_transformer = Arc::new(f);
+        self
+    }
+
+    /// Proxy config for tcp
+    #[cfg(not(target_family = "wasm"))]
+    pub fn tcp_proxy_config(mut self, proxy_url: Option<String>) -> Self {
+        self.config.tcp_config.tcp.proxy_url = proxy_url;
+        self
+    }
+
+    /// Onion config for tcp
+    #[cfg(not(target_family = "wasm"))]
+    pub fn tcp_onion_config(mut self, onion_url: Option<String>) -> Self {
```

Comment:
if set proxy or onion, it must be a string. The option is inner struct

Reply:
Yes.